cluster-operator/src/main/java/io/strimzi/operator/cluster/model/JmxTrans.java

```diff
@@ -0,0 +1,364 @@
+                .withResources(getResources())
+                .withVolumeMounts(getVolumeMounts())
+                .withImagePullPolicy(determineImagePullPolicy(imagePullPolicy, getImage()))
+                .build();
+
+        containers.add(container);
+
+        return containers;
+    }
+
+    @Override
+    protected List<EnvVar> getEnvVars() {
+        List<EnvVar> varList = new ArrayList<>();
+
+        if (isJmxAuthenticated()) {
+            varList.add(buildEnvVarFromSecret(KafkaCluster.ENV_VAR_KAFKA_JMX_USERNAME, KafkaCluster.jmxSecretName(cluster), KafkaCluster.SECRET_JMX_USERNAME_KEY));
+            varList.add(buildEnvVarFromSecret(KafkaCluster.ENV_VAR_KAFKA_JMX_PASSWORD, KafkaCluster.jmxSecretName(cluster), KafkaCluster.SECRET_JMX_PASSWORD_KEY));
+        }
+
+        addContainerEnvsToExistingEnvs(varList, Collections.emptyList());
+
+        return varList;
+    }
+
+    /**
+     * Generates the name of the JmxTrans deployment
+     *
+     * @param kafkaCluster  Name of the Kafka Custom Resource
+     * @return  Name of the JmxTrans deployment
+     */
+    public static String jmxTransName(String kafkaCluster) {
+        return JmxTransResources.deploymentName(kafkaCluster);
+    }
+
+    /**
+     * Get the name of the JmxTrans service account given the name of the {@code kafkaCluster}.
+     * @param kafkaCluster The cluster name
+     * @return The name of the JmxTrans service account.
+     */
+    public static String containerServiceAccountName(String kafkaCluster) {
+        return JmxTransResources.serviceAccountName(kafkaCluster);
+    }
+
+    @Override
+    protected String getDefaultLogConfigFileName() {
+        return null;
+    }
+
+    @Override
+    protected String getServiceAccountName() {
+        return JmxTransResources.serviceAccountName(cluster);
+    }
+
+    public static String jmxTransConfigName(String cluster) {
+        return cluster + JMX_METRICS_CONFIG_SUFFIX;
+    }
+
+    public boolean isDeployed() {
+        return isDeployed;
+    }
+
+    public boolean isJmxAuthenticated() {
+        return isJmxAuthenticated;
+    }
+
+    /**
+     * Wrapper class used to create the overall config file
+     * Servers: A list of servers and what they will query and output
+     */
+    @SuppressWarnings("SE_BAD_FIELD_INNER_CLASS")
+    static class Servers implements Serializable {
+        private static final long serialVersionUID = 1L;
+        public List<Server> servers;
+    }
+
+    /**
+     * Wrapper class used to create the per broker JmxTrans queries and which remote hosts it will output to
+     * host: references which broker the JmxTrans will read from
+     * port: port of the host
+     * queries: what queries are passed in
+     */
+    @SuppressWarnings("SE_BAD_FIELD_INNER_CLASS")
+    static class Server implements Serializable {
+        private static final long serialVersionUID = 1L;
```

Comment:
Could you please give these classes a bit better formatting? New lines between the fields etc., to make it a bit more readable?

Reply:
Will do!